src/v/storage/disk_log_impl.cc

```diff
@@ -1027,23 +1027,26 @@ ss::future<> disk_log_impl::housekeeping(housekeeping_config cfg) {
      */
     if (config().is_compacted() && !_segs.empty()) {
         if (config::shard_local_cfg().log_compaction_use_sliding_window()) {
-            co_await sliding_window_compact(cfg.compact, new_start_offset);
-            if (auto range = find_compaction_range(cfg.compact); range) {
-                auto r = co_await compact_adjacent_segments(
-                  std::move(*range), cfg.compact);
-                vlog(
-                  gclog.debug,
-                  "Adjacent segments of {}, compaction result: {}",
-                  config().ntp(),
-                  r);
-                if (r.did_compact()) {
-                    _compaction_ratio.update(r.compaction_ratio());
+            auto compacted = co_await sliding_window_compact(
+              cfg.compact, new_start_offset);
+            if (!compacted) {
```

Comment:
> storage: only merge adjacent segments if didn't window compact

how does this relate to the comment a few commits back that adjacent segment merging is needed because of the assumptions made by so many tests that the segment count is reduced by the compaction process.

is the idea here with this change that adjacent merging just waits until window comapction is fully complete?

Reply:
Sorry, I'll expand the commit message for this but it is related to those same tests. This is a hack to ensure there's a stage in between sliding window compaction and adjacent merge compaction where all original segments are still present but compacted fully. A number of tests expect such a stage

Reply:
Done, merged this commit with the one you noted here and updated the commit message